Makefile

```diff
@@ -60,10 +60,9 @@ test-compile-changed: os-autoinst/
 test-yaml-valid:
 	@# Get list of changed yaml files. We only want to lint changed files for
 	@# now because yamllint complains about a lot of existing files
-	$(eval YAMLS=$(shell sh -c "git ls-files schedule/ test_data/ | grep '/.*\.y.\?ml$$'"))
-	export PERL5LIB=${PERL5LIB_} ; tools/test_yaml_valid $(YAMLS);\
-	$(eval CHANGED_YAMLS=$(shell sh -c "git --no-pager diff --diff-filter=dr --name-only master | grep '\(schedule\|test_data\)/.*\.y.\?ml$$'"))
+	$(eval CHANGED_YAMLS=$(shell sh -c "git --no-pager diff --diff-filter=dr --name-only FETCH_HEAD `git merge-base FETCH_HEAD origin/master` | grep '\(schedule\|test_data\)/.*\.y.\?ml$$'"))
```

Comment:
maybe you could also change the `y.\?ml` as practically it will match also yeml,y222ml etc
Also from the man page
```
git diff [<options>] <commit>...<commit> [--] [<path>...]
           This form is to view the changes on the branch containing and up to the second <commit>,
           starting at a common ancestor of both <commit>. "git diff A...B" is equivalent to "git
           diff $(git merge-base A B) B". You can omit any one of <commit>, which has the same effect
           as using HEAD instead.
``` seems simplier

Reply:
I will focus on making it work first, I cannot improve all the problems we have in Makefile in 5 minutes. Feel free to create separate PR with this change.

Reply:
As for diff option, I've tested it and indeed we can put branch name, so will simplify that part.